lib/_http_client.js

```diff
@@ -629,7 +630,7 @@ function responseOnEnd() {
         socket.end();
     }
     assert(!socket.writable);
-  } else if (req.finished) {
+  } else if (req.finished && !this.aborted) {
```

Comment:
If response is `aborted` then the socket is closed and there is nothing to keep alive.